### PR TITLE
[v6r17] Improvement in getFilesToStage, JobScheduling snd TS 

### PR DESCRIPTION
**Pull request by phicharp**
Replaces https://github.com/DIRACGrid/DIRAC/pull/3294

* getFilesToStage was only returning online files
* JobScheduling was assuming all non-online files required staging
* Improvements include:
return explicitly offline files and only stage those if at TpeSE
return also absent files (i.e. not at SE): in this case make the job Failed
* Improvements in TaskManager: process tasks in chunks of 100 in order to update faster the TS (tasks and files)

### CODE REVIEW

**Comment by phicharp** [comment]
I added a stupid piece of code in order top make tests happy, as am_getOption cannot return anything sensible with just a mock...
This should be fixed by an expert, I don't know how to do it

File: StorageManagementSystem/Client/StorageManagerClient.py [hunk 1]
```diff
@@ -105,12 +119,15 @@ def _checkFilesToStage( seToLFNs, onlineLFNs ):
...
+        if cmpError( reason, errno.ENOENT ):
+          absentLFNs.setdefault( lfn, [] ).append( se )
+          failed[se].pop( lfn )
     if not failed[se]:
       failed.pop( se )
+  # Find the files that do not exist at SE
   if failed:
-    gLogger.error( "Could not get metadata", "for %d files" % len( set( lfn for lfnList in failed.itervalues() for lfn in lfnList ) ) )
-    return S_ERROR( "Could not get metadata for files" )
+    gLogger.error( "Error getting metadata", "for %d files" % len( set( lfn for lfnList in failed.itervalues() for lfn in lfnList ) ) )
```
Comment (fstagni) [inline StorageManagementSystem/Client/StorageManagerClient.py]: For my understanding: why the `return S_ERROR` was removed?

**Comment by phicharp** [comment]
because there may be successful and failed, this is handled in the calling code rather than returning an error
In fact that method doesn't need to return anything as it only returns S_OK()

**Comment by phicharp** [comment]
Ah not really... If RSS is not responding we give up

**Pull request merged**